nsqd/nsqd.go

```diff
@@ -425,19 +425,20 @@ func (n *NSQD) PersistMetadata() error {
 	// if no symlink (yet), race condition:
 	// crash right here may cause next startup to see metadata conflict and abort
 
+	// on Windows, Administrator privs are needed to Symlink, so skip rollback compat
+	if runtime.GOOS == "windows" {
+		if err == nil || !os.IsNotExist(err) {
+			err = os.Remove(fileNameID)
+		}
+		return err
+	}
```

Comment:
That's a bug, I was trying to be too concise, nice catch.
Looks like I'll back-out this windows-path variant anyway.